.circleci/initjob/40-set-label-environment-variables.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+shared_env=$1
+
+if [[ -f pr-metadata/labels/build-legacy-probes ]]; then
+    echo 'export BUILD_LEGACY_PROBES=true' >> "$shared_env"
```

Comment:
I'm a little confused around the difference between `$BASH_ENV` and `$SHARED_ENV`. We're writing a bunch of environment variables into `$SHARED_ENV` (which is `${WORKSPACE_ROOT}/shared-env` and then copying the contents of that into `$BASH_ENV` which is defined by circle ci. We also have scripts that write directly into `BASH_ENV`. Is there any reason to keep `shared-env` at all if it all ends up in the other location? I think consistency is key here so either all our scripts should write to `shared-env` and then its copied (or sourced on demand?) or everything should write to `$BASH_ENV` directly

Reply:
$BASH_ENV is not saved from job to job whereas $SHARED_ENV is. This is a good source to learn about it https://medium.com/@johnthughes/circleci-persisting-environment-variables-across-steps-and-jobs-5276670cf590